spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourceEntitlementListSpecTest.java

```diff
@@ -0,0 +1,104 @@
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.invoker.client.ApiException;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.ProductAttributes;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+@SpecTest
+public class OwnerResourceEntitlementListSpecTest {
+
+    private static ApiClient adminClient;
+    private static ConsumerDTO consumer;
+
+    @BeforeAll
+    public static void setUp() throws ApiException, JsonProcessingException {
+        adminClient = ApiClients.admin();
+
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        ProductDTO monitoringProduct =
+            createProduct(adminClient, owner, ProductAttributes.Variant.withValue("Satellite Starter Pack"));
+        ProductDTO virtualProduct = adminClient.ownerProducts().createProductByOwner(
+            owner.getKey(), Products.random());
+
+        // entitle owner for the virtual and monitoring products
+        PoolDTO pool1 = adminClient.owners().createPool(owner.getKey(), Pools.random(monitoringProduct));
+        PoolDTO pool2 = adminClient.owners().createPool(owner.getKey(), Pools.random(virtualProduct));
+
+        UserDTO user = UserUtil.createUser(adminClient, owner);
+        ApiClient userClient = ApiClients.basic(user.getUsername(), user.getPassword());
+        consumer = userClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        consumerClient.consumers().bindPool(consumer.getUuid(), pool1.getId(), 1);
+        consumerClient.consumers().bindPool(consumer.getUuid(), pool2.getId(), 1);
+    }
+
+    @Test
+    public void shouldFetchAllEntitlementsOfAnOwner() throws ApiException {
+        List<EntitlementDTO> ents = adminClient.consumers()
+            .listEntitlements(consumer.getUuid(), null, false, null, null, null, null, null);
+        assertEquals(2, ents.size());
+    }
+
+    @Test
+    public void shouldAllowFilterConsumerEntitlementsByProductAttribute() throws ApiException {
+        List<EntitlementDTO> ents = adminClient.consumers().listEntitlements(
+            consumer.getUuid(), null, false, List.of("variant:Satellite Starter Pack"),
+            null, null, null, null);
+        assertEquals(1, ents.size());
+
+        List<AttributeDTO> attributes = ents.stream().findFirst()
```

Comment:
This is an interesting way of doing this, but given that we're testing the whole chain all the way down, this can be done with an assertion chain instead:

```java
assertThat(ents)
  .isNotNull()
  .singleElement()
  .extracting(EntitlementDTO::getPool)
  .isNotNull()
  .extracting(PoolDTO::getProductAttributes, as(collection(AttributeDTO.class)))
  .isNotNull()
  .singleElement()
  .returns("variant", AttributeDTO::getName)
  .returns("Satellite Starter Pack", AttributeDTO::getValue);
```

That way we're verifying the output of our target method down the whole tree, and any issue or incorrect output will be verified, or throw an assertion error rather than an unasserted exception.

Reply:
done

Reply:
The old attributes checks still look to be present below them. With this chain in place, the other checks are no longer necessary.

Reply:
done